sno/merge.py

```diff
@@ -48,18 +76,29 @@ def merge(ctx, ff, ff_only, dry_run, commit):
     # accept ref-ish things (refspec, branch, commit)
     theirs = CommitWithReference.resolve_refish(repo, commit)
     ours = CommitWithReference.resolve_refish(repo, "HEAD")
-
-    click.echo(f"Merging {theirs} to {ours} ...")
     ancestor_id = repo.merge_base(theirs.id, ours.id)
-    click.echo(f"Found common ancestor: {ancestor_id}")
 
     if not ancestor_id:
         raise InvalidOperation(f"Commits {theirs.id} and {ours.id} aren't related.")
 
+    merge_message = f"Merge {theirs.shorthand_with_type} into {ours.shorthand}"
+    merge_jdict = {
+        "ancestor": {"commit": ancestor_id.hex},
+        "ours": ours.as_json(),
+        "theirs": theirs.as_json(),
+        "message": merge_message,
+        "conflicts": None,
+    }
+
     # We're up-to-date if we're trying to merge our own common ancestor.
     if ancestor_id == theirs.id:
-        click.echo("Already merged!")
-        return
+        merge_jdict["mergeCommit"] = ours.id.hex
+        merge_jdict["no-op"] = True
```

Comment:
I feel weird about this hyphen when other things are camelCase

Reply:
no-op -> noOp